consensus/dummy/consensus.go

```diff
@@ -41,7 +40,31 @@ type Mode struct {
 }
 
 type (
+	OnFinalizeAndAssembleCallbackType = func(
```

Comment:
It looks like these changes are related to the original PR we're syncing here. Is there a reason they need to be included in this PR?

Reply:
Yeah totally understand that - if syncing a coreth PR requires bringing in other code (that wasn't brought in the past, I've done it in the same PR (as otherwise would sacrifice speed). But yes generally understand and agree with what you're saying. If you'd like me to separate these out into two PRs I certainly can.